src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -205,6 +206,13 @@ export class MockHttpSocket extends MockSocket {
       return
     }
 
+    // Return fake address information for the socket.
+    this.address = () => ({ 
+      address: '0.0.0.0',
+      family: 'IPv4',
```

Comment:
I wonder if there's any info in `connectionOptions` that we can use to set the `family`? 🤔 

Yeah, there's `.family` and `.host`. Do you think we should respect those?

```ts
this.address = () => {
  return {
    address: this.connectionOptions?.host || '127.0.0.1',
    family: this.connectionOptions?.family || 'IPv6',
    port: this.connectionOptions?.port || 0
  }
}
```

We can also add automated tests for the cases when explicit connection options are set. Granted, this is likely better tested in the unit tests on `MockHttpSocket`. You can leave the existing `http.*` tests as-is, those are useful too!

Reply:
I think the address info resolved in the socket. I didn't verfiy this, but the `connectionOptions` includes only the `port`.
Also, please note the `host` !== `address`.

Reply:
You're right. Let's infer the `port` only for now then?